add integ-calico job

ilrudie opened the pull request:
Adds an ambient test for Calico CNI as suggested [here](https://github.com/istio/istio/pull/49255#issuecomment-1932467482).

This addition will be used to help make sure that our inpod capture method functions as expected with Calico. Other CNIs may be considered/added later as well as tests which rely on these CNIs for configurations which aren't possible with the basic KinD CNI (example, K8s Network Policy).

PRs this depends on:

- https://github.com/istio/istio/pull/49255
- https://github.com/istio/common-files/pull/990

ilrudie committed:
add integ-calico job

Signed-off-by: ilrudie <ian.rudie@solo.io>

dhawton commented on prow/config/jobs/istio.yaml:
```diff
@@ -76,6 +76,20 @@ jobs:
       - name: INTEGRATION_TEST_FLAGS
         value: --istio.test.skipVM
 
+  - name: integ-ambient-calico
```
```suggestion
  - name: integ-ambient-calico
    modifiers: [presubmit_optional]
```

I think it should be optional until we know it's working as expected, then we can make it required.. as well as any other bugs that may need addressing without full stopping other work.

ilrudie replied:
Sounds like a good plan. TY

ilrudie committed:
Update prow/config/jobs/istio.yaml

Make the new job optional at first to build confidence before it becomes required.

Co-authored-by: Daniel Hawton <daniel@hawton.org>

ilrudie committed:
make gen for dhawton's suggestion

Signed-off-by: ilrudie <ian.rudie@solo.io>

howardjohn commented on prow/config/jobs/istio.yaml:
```diff
@@ -76,6 +76,21 @@ jobs:
         value: --istio.test.skipVM
 
+  - name: integ-ambient-calico
+    modifiers: [presubmit_optional]
```
I don't think we should run this on presubmit. Its a useful test but IMO we don't want to bloat up our test matrix this much (esp since we will be adding cilium, ..., in the future). can we make it `[presubmit_optional, presubmit_skipped]`?

ilrudie replied:
makes sense. thanks for the suggestion

ilrudie committed:
adding presubmit_skipped modifier to new job

Signed-off-by: ilrudie <ian.rudie@solo.io>

ilrudie committed:
run make gen

Signed-off-by: ilrudie <ian.rudie@solo.io>
